### DISCUSSION TITLE
2024-07-24 수요일

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#90] [기록보드]
2024-07-24 수요일 기록보드입니다.

## 여행가고십조(A10) 일일 공통 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 일일 1차 정기 회의 | |
| 10:00 ~ 10:30 | 하루 일과 준비 | |
| 10:30 ~ 13:00 | 개발 | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 15:00 | 일일 2차 정기 회의 | |
| 15:00 ~ 18:00 | 개발 | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 일일 3차 정기 회의, 튜터님 질의응답(동적 헤더) | <img src="https://github.com/user-attachments/assets/c8b9b455-3140-4964-90c8-124ca88e3707" weight="300" height="300"/> |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

```
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 본인인증 외부 API 응답값 확인 및 테스트

// 사진
<img src="이미지 주소" weight="300" height="300"/>

### 트러블 슈팅

- user id를 uuid로 받다 보니 segment가 지저분 해짐 -> login_id를 추가로 받는 것으로 해결

### 내일 계획

- supabase auth 비밀번호 찾기 기능 연결
```

**Comment by Captain-Kim** [comment]
팀 공통 튜터님께 전달드린 내용

**오늘 공통 일과표 (2024-07-24)**
https://github.com/nbc-final-a10/TripBuddy/discussions/90

- 09 : 00 ~ 09 : 30 오전 안부 인사
- 10 : 00 ~ 13 : 00 개발
- 14 : 00 ~ 15 : 00 오후 회의
- 15 : 00 ~ 18 : 00 개발
- 19 : 00 ~ 20 : 00 저녁 회의

**오늘 예정 사항**

- 재사용을 고려한 컴포넌트 작성 (예를 들어 검색과 글쓰기에서 겹치는 컴포넌트는 UI 작성은 현정, 컴포넌트 분리는 병준)

**어제 회의 주요 내용**
https://github.com/nbc-final-a10/TripBuddy/discussions/83
- 디자이너님께 전달할 내용
  - 튜토리얼 페이지 와이어프레임 없음 -> 협의
  - 인증/인가 와이어프레임 없음 -> 엔지니어 측에서 전달
  - 메인 페이지 아이템 리스트 가로 스크롤 우측 패딩 적용해야 할 듯
  - 검색 페이지에서 추가 속성 선택에 대한 UI가 없음 -> 게시글 작성 페이지에서 컴포넌트 재사용 제안
  - 튜터님 피드백 사항 적극 반영 (컴포넌트 재사용, 코드 퀄리티 향상 제고)
- 재사용 가능한 컴포넌트 찾기
  - 검색 페이지 <-> 글쓰기 페이지 3개 페이지에서 컴포넌트 겹침 (chips, 달력, 인원 선택)
- TailwindCSS 스타일링 방법(인라인 등), useState 등의 React hooks에 타입 지정하기 등 일관성 있는 코드 작성
- 최종 프로젝트 지원금 관련
  - FE, BE 유료 결제 중이나 지원금 사용할 의사 없음 -> 팀원들께서 비용을 사용해서 퀄리티를 높일 수 있는 부분이 있다면 공유해주시고 -> 없으면 UT 비용으로 전액 사용
  - UT는 프로젝트 진행하면서 해보기 쉽지 않은 경험일 것. 양질의 피드백을 수렴하는 것이 우리에게도 도움이 될 듯.
- 온보딩 넘기기 기능 추가
  - isOnboading 컬럼의 기본값이 false인데 온보딩 skip 버튼을 누르면 true로 바꿔서  온보딩을 스킵하게 구현할 수 있다 -> 그러나 데이터 테이블 컬럼의 의미와는 다른 것 같아 아직 프로젝트에서 수파베이스를 본격적으로 사용하지 않았으니 isSkip이라는 컬럼을 추가하자
프로필 사진은 온보딩 시 생략 가능하도록 구현 -> 생략 시 기본 프사 또는 provider 제공 프사
- 달력 구현 방법은 NextUI 사용
  - 해당 라이브러리에 더 좋은 기능이 많지만 우리 팀은 라이브러리 의존도를 최소화하는 것을 지향 -> 달력과 같이 복잡한 것만
  - 아이콘 등의 assets은 프로젝트에, 수파베이스 서버에? -> FE (static assets은 FE가 나은 경험제공할 듯)

**공통 트러블 슈팅**
없음

**Comment by eunohhh** [comment]
| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 일일 1차 정기 회의 |  안부확인 |
| 10:00 ~ 10:30 | 하루 일과 준비 | |
| 10:30 ~ 13:00 | 개발 | Chip을 위주로 진행 |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 15:00 | 일일 2차 정기 회의 | |
| 15:00 ~ 18:00 | 개발 | onBoarding을 위주로 진행 |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 일일 3차 정기 회의 | |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

### 오늘 한 일

- 인증/인가/온보딩 첫 PR
- 온보딩관련 api 작성

### 트러블 슈팅

- Chip 재사용을 위해 cva 라이브러리로 변경(O_Chip.tsx)

### 내일 계획

- 온보딩 페이지 다듬기!

**Comment by slsl2** [comment]
## 개인 기록보드 (김민지)


### 오늘 한 일

- 메인페이지, 탭메뉴 컴포넌트 보다 더 세분화, 재사용성 고려
- 헤더, 탭메뉴 ui 수정
- 이번 프로젝트 첫 PR, 첫 merge
- chat 페이지 (채팅 목록) 스켈레톤 작성
- chat 페이지 구현 시 레이아웃 구성 관련 고민, 레퍼런스 서치


### 내일 계획

- chat 페이지 레이아웃 구성 작업 (모바일과 데스크탑 모두 고려)
- chat 페이지 리스트 컴포넌트 ui 작업
- chat 상세 페이지 ui 작업 시작

**Comment by hyunjeongkwak** [comment]
**오늘 한 일**

- 날짜 선택 페이지 구현(달력 라이브러리 사용)

**내일 할 일**

- 검색 페이지 내에 스크롤, 퍼널 등 재사용 가능한 컴포넌트 바꿔주기(진행 중)
- 검색 결과 페이지 UI 제작

(내일 모레 할 일)

- 와이어프레임 결정되었다면,
- 검색메인, 위치검색, 날짜 선택 페이지 1280px UI 제작

**Comment by Captain-Kim** [comment]
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 목적지 선택 컴포넌트 atoms 단위로 재사용 가능하도록 분해
- 모의 기술면접

### 트러블 슈팅

- reusable components로 리팩토링하면서 스타일링이 깨지는 등 어려움을 겪고 있음. -> 하나씩 잡아 가는 중

### 내일 계획

- 날짜선택 컴포넌트 작성 완료되면 해당 페이지 분해 및 글쓰기 페이지 제작
